src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int newDefaultCapacity = DEFAULT_CAPACITY;
+    private T[] elementData;
+    private int size = 0;
+    
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+    
+    public void grow(int size) {
+        if (size == newDefaultCapacity) {
+            newDefaultCapacity = (int) (elementData.length * 1.5);
+            elementData = Arrays.copyOf(elementData, newDefaultCapacity);
+        }
+    }
+    
+    public void indexOutOfBounds(int index) {
+        throw new ArrayListIndexOutOfBoundsException("Index " + index + " out of bounds array");
+    }
+    
     @Override
     public void add(T value) {
-
+        grow(size);
```

Comment:
name `grow()` means that method should grown every time, but it's not true
```suggestion
        checkSizeAndGrow(size);
```
or
```suggestion
        growIfNeed(size);
```